src/client/app.ts

```diff
@@ -29,24 +31,39 @@ class MazeGame {
         const element: HTMLElement = document.getElementById("mazeArea");
         element.innerHTML = "<canvas id=\"maze\" style=\"border:1px solid #000000\"></canvas>";
 
-        const canvas: HTMLCanvasElement = document.getElementById("maze") as HTMLCanvasElement;
-        canvas.width = canvasWidth;
-        canvas.height = canvasHeight;
+        //const canvas: HTMLCanvasElement = document.getElementById("maze") as HTMLCanvasElement;
+        this.canvas = document.getElementById("maze") as HTMLCanvasElement;
+        this.canvas.width = canvasWidth;
+        this.canvas.height = canvasHeight;
 
         this.maze = maze;
-        this.mazeDrawer = new MazeDrawer(this.maze, canvas);
+        this.mazeDrawer = new MazeDrawer(this.maze, this.canvas);
         this.mazeDrawer.drawMaze();
         this.mazePath = new MazePath(this.mazeDrawer);
         this.mazePath.add(this.maze.getCell(0, 0));
 
-        canvas.addEventListener("mousemove", (event: MouseEvent) => {
-            this.handleMouseDrag(canvas, event);
+        this.canvas.addEventListener("mousemove", (event: MouseEvent) => {
+            this.handleMouseDrag(this.canvas, event);
         });
 
         this.isMazeSolved = false;
         this.startTime = new Date();
     }
 
+    solveMaze() {
+        if (!this.isMazeSolved) {
+            const finishTime: Date = new Date();
+            const timeDelta: number = finishTime.getTime() - this.startTime.getTime();
+
+            this.isMazeSolved = true;
+            this.mazePath = getSolution(this.maze, new MazePath(new MazeDrawer(this.maze, this.canvas)));
```

Comment:
I don't know if `getSolution` should return the solution MazePath -- what is the reason for updating `this.mazePath`?